### DISCUSSION TITLE
How to swipe between tabs

### DISCUSSION THREAD

**Discussion by MoeHamdan** [roubachof/Sharpnado.Tabs#136] [General]
**Is your feature request related to a problem? Please describe.**
I want the  TabHostView  and ViewSwitcher to have the carousel effect? In other words make it work just like a tabstrip works on Android.[...]

Thank you very much for this amazing library.

**Comment by roubachof** [comment]
You can already :)
Use a TabHostView with the HorizontalList (with Carousel Layout) instead of the ViewSwitcher, and bind SelectedIndex to CurrentIndex.

**Comment by MoeHamdan** [comment]
Oh! You are awesome!!! That worked perfectly!

I have 1 more small question, how would I make the underlined part of the tab header to have the carousel effect? is that possible?

**Comment by roubachof** [comment]
unfortunately it would require a new implementation of the tabs for this...

**Comment by MoeHamdan** [comment]
I see thank you! I will try implementing the tab.

**Comment by roubachof** [comment]
You can create a feature request for this.
I will have time to improve tabs this summer :)

**Comment by MoeHamdan** [comment]
If I was successful I will pull request it. If not I will feature request it.

Again, Than you so much for this library!

**Comment by TienNK1987** [comment]
> You can already :)
> Use a TabHostView with the HorizontalList (with Carousel Layout) instead of the ViewSwitcher, and bind SelectedIndex to CurrentIndex.

Hi @roubachof, 
How to use HorizontalList (with Carousel Layout) instead of ViewSwitcher with different content in each tab? As I know, HorizontalListView has only one DataTemplate to declare its content.

Thank you for your amazing lib.

**Comment by roubachof** [comment]
With a DataTemplateSelector :)

**Comment by TienNK1987** [comment]
> With a DataTemplateSelector :)
Could you please give me a sample?
Thank you @roubachof,

**Comment by roubachof** [comment]
https://docs.microsoft.com/en/xamarin/xamarin-forms/app-fundamentals/templates/data-templates/selector

**Comment by dpjha84** [comment]
Shouldn't this be the default behaviour. If not, is it planned in the future?

**Comment by roubachof** [comment]
Use a TabHostView with the [HorizontalList](https://github.com/roubachof/Sharpnado.CollectionView#carousel-layout) (or even with [CardsView](https://github.com/AndreiMisiukevich/CardView.MAUI) instead of the ViewSwitcher, and bind SelectedIndex to CurrentIndex.

**Comment by eyeveye** [comment]
If we use the horizontalist, can we still use the delayed view? I found the delayed view is very useful.

**Comment by mpapadis** [comment]
I'm looking to divide a view into tabs, allowing users to navigate without scrolling by organizing properties into context-specific tabs (for instance, one tab for credentials and another for notification preferences in a profile view). As a collection view isn't suitable for this, I'd like to implement a way to swipe between tabs. I've experimented with SwipeGestureRecognizers, but the outcome wasn't ideal.
